src/deploy/functions/validate.ts

```diff
@@ -85,3 +79,26 @@ export function packageJsonIsValid(
     throw new FirebaseError(msg);
   }
 }
+
+function _functionsSourceIsPresent(data: any, sourceDir: string, projectDir: string): void {
```

Comment:
I'd also recommend: rename this function so it's obvious it will do Something Bad if it fails whatever check. Either `validateFunctionsSourceIsPresent` or `assertFunctionsSourceIsPresent` are recommendations off the top of my head.

(This applies to `enginesFieldIsPresent` as well)

Reply:
Done.